src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1115,6 +1115,26 @@ private static void ExpandTypeName(TypeSyntax type, StringBuilder builder)
                     builder.Append(qualified.DotToken.Text);
                     ExpandTypeName(qualified.Right, builder);
                     break;
+                case SyntaxKind.TupleType:
```

Comment:
So, to fix this crash, i'm fine with your change.  But i think i'm going to rip all this code out anyways since it seems like *not at all* what we want.  DeclaredSymbolInfo is for storing structured syntactic information about a symbol.  Munging things into opaque strings is not at all what we want to use it for :)